packages/ui/src/SegmentedControl.tsx

```diff
@@ -4,48 +4,90 @@ import {Box} from "./Box";
 import {SegmentedControlProps} from "./Common";
 import {Text} from "./Text";
 
-export class SegmentedControl extends React.Component<SegmentedControlProps, {}> {
-  renderItem(item: string | React.ReactNode) {
+export const SegmentedControl = ({
+  items,
+  onChange = () => {},
+  selectedItemIndexes = undefined,
+  selectedItemIndex = undefined,
+  multiselect = false,
+  selectLimit = 1,
+}: SegmentedControlProps) => {
+  const renderItem = (item: string | React.ReactNode) => {
     return <Text weight="bold">{item}</Text>;
     // if (typeof item === "string") {
     //   return <Text weight="bold">{item}</Text>;
     // } else {
     //   return item;
     // }
+  };
+
+  if (selectedItemIndex === undefined && selectedItemIndexes === undefined) {
+    console.warn("One of the following must be defined: selectedItemIndex, selectedItemIndexes");
+    return null;
+  }
+
+  if (!multiselect && selectedItemIndexes?.length && selectedItemIndexes?.length > 1) {
+    console.warn("Muliple selections not allowed without multiselect flag");
+    return null;
   }
 
-  render() {
-    return (
-      <Box
-        color="lightGray"
-        direction="row"
-        display="flex"
-        height={40}
-        padding={1}
-        rounding={3}
-        width="100%"
-      >
-        {this.props.items.map((item, index) => (
+  if (selectedItemIndexes?.length && selectedItemIndexes?.length > selectLimit) {
+    console.warn("The number of selected items exceeds the limit");
+    return null;
+  }
+
+  const isTabActive = (index: any) => {
+    return selectedItemIndex === index || selectedItemIndexes?.includes(index)
+      ? "white"
+      : "lightGray";
+  };
+
+  return (
+    <Box
+      color="lightGray"
+      direction="row"
+      display="flex"
+      height={40}
+      justifyContent="between"
+      padding={1}
+      rounding={3}
+      width="100%"
+    >
+      {items.map((item, index) => (
+        <Box
+          key={index}
+          color={isTabActive(index)}
+          height="100%"
+          paddingX={2}
+          rounding={3}
+          // The additional 10% missing will act as padding between controls
+          width={`${90 / items.length}%`}
```

Comment:
I don't think we should do this, and leave it up to the children to decide how much padding they need (e.g. wrapped in a top level Box with paddingX=4 or something). In some cases I could imagine consumers wanting to have them close or touching.